libs/ml/src/ops/activations/gelu.cpp

```diff
@@ -0,0 +1,146 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2020 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "math/activation_functions/gelu.hpp"
+#include "math/trigonometry.hpp"
+#include "ml/ops/activations/gelu.hpp"
+
+namespace fetch {
+namespace ml {
+namespace ops {
+
+template <typename TensorType>
+Gelu<TensorType>::Gelu(SPType const &sp)
+  : Ops<TensorType>(sp)
+{}
+
+template <typename TensorType>
+std::shared_ptr<OpsSaveableParams> Gelu<TensorType>::GetOpSaveableParams()
+{
+  auto sp = std::make_shared<SPType>();
+  return sp;
+}
+
+template <typename TensorType>
+std::shared_ptr<fetch::ml::ops::Ops<TensorType>> Gelu<TensorType>::MakeSharedCopy(
+    std::shared_ptr<fetch::ml::ops::Ops<TensorType>> me)
+{
+  FETCH_UNUSED(me);
+  assert(me.get() == this);
+
+  auto copyshare = std::make_shared<MyType>(*this);  // calls default copy constructor of MyType
+
+  return copyshare;
+}
+
+// 0.5x(1+tanh(0.797885x+0.035677x^3))
```

Comment:
I guess it would be great to have some more explanation about what does this formula mean and why is it used.

Reply:
ill add a reference to the paper